vllm/v1/spec_decode/eagle.py

```diff
@@ -667,8 +701,13 @@ def load_model(self, target_model: nn.Module) -> None:
     def dummy_run(
         self,
         num_tokens: int,
+        attn_metadata: Optional[dict[str, Any]] = None,
     ) -> None:
-        with set_forward_context(None, self.vllm_config,
+        if attn_metadata is not None and self.attn_metadata_cudagraph is None:
+            self.attn_metadata_cudagraph = attn_metadata[
+                self.attn_layer_names[0]]
```

Comment:
QQ: Why 0-th layer?

Reply:
All draft attention layers share the same attn_metadata (line 174-184), so using any one of the attn_layer_names would work here. Lemme add comment to clarify this

Reply:
Added comment here